lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -41,6 +41,21 @@ def validate_discovery_opts
         unless @discovery['hosts']
       raise ArgumentError, "invalid zookeeper path for service #{@name}" \
         unless @discovery['path']
+      if @discovery['decode']
+        raise ArgumentError, "missing or invalid decode method #{@discovery['decode']}" \
+          unless @discovery['decode']['method']
+        if @discovery['decode']['method'] == 'aurora'
```

Comment:
I feel like this code should live in the constructor and be factored out into a function e.g. parse_decode_options or something. Feel free to go "meh" and I'll do it in a follow up.